modern/src/reports/ReplayPage.js

```diff
@@ -40,14 +61,42 @@ const TimeLabel = ({ children, open, value }) => (
 );
 
 const ReplayPage = () => {
-  const classes = useStyles();
   const t = useTranslation();
+  const classes = useStyles();
+  const history = useHistory();
+  const timerRef = useRef();
 
-  const [expanded, setExpanded] = useState(true);
   const [positions, setPositions] = useState([]);
   const [index, setIndex] = useState(0);
+  const [selectedDeviceId, setSelectedDeviceId] = useState();
+  const [playbackSpeed, setPlaybackSpeed] = useState('');
+  const [expanded, setExpanded] = useState(true);
+  const [isPlaying, setIsPlaying] = useState(false);
+
+  const deviceName = useSelector((state) => {
+    if (selectedDeviceId) {
+      const device = state.devices.items[selectedDeviceId];
+      if (device) {
+        return device.name;
+      }
+    }
+    return null;
+  });
+
+  useEffect(() => {
+    if (isPlaying && positions.length > 0) {
+      timerRef.current = setInterval(() => {
```

Comment:
I can't seem to find a place where you stop timer after reaching the end. Did I miss something?

Reply:
Done